compiler/src/dotty/tools/dotc/cc/Setup.scala

```diff
@@ -454,7 +454,7 @@ class Setup extends PreRecheck, SymTransformer, SetupAPI:
           case _ => false
 
         def signatureChanges =
-          tree.tpt.hasRememberedType && !sym.isConstructor || paramSignatureChanges
+          (tree.tpt.hasRememberedType || tree.tpt.isInstanceOf[InferredTypeTree]) && !sym.isConstructor || paramSignatureChanges
```

Comment:
Why is the added condition needed?

Reply:
After a second thought this feels like a hack. The actual problem this tries to solve is that, for instance, given a ValDef `value x: A^{y}`, the PostTyper cleans up the capture set in the type tree of the definition but the `info` of the symbol `x` is not updated.

Later on `Setup` tries to read the info of that symbol [here](https://github.com/scala/scala3/blob/447cdf48ef7f2cae11118ad9b31d7e62c2da3e43/compiler/src/dotty/tools/dotc/cc/Setup.scala#L98), and if the capture set is invalid, the compiler crashes.